src/classes/Policy.js

```diff
@@ -174,6 +175,34 @@ class Policy {
 		}
 		return { block: false, reason: allowedOnce ? BLOCK_REASON_C2P_ALLOWED_ONCE : BLOCK_REASON_GLOBAL_UNBLOCKED };
 	}
+
+	/**
+	 * Check given url against pattern which might be a wildcard, or a regex
+	 * @param  {string} url		site url
+	 * @param  {string} pattern	regex pattern
+	 * @return {boolean}
+	 */
+	matchesWildcardOrRegex(url, pattern) {
+		// Pattern might be a wildcard
+		if (pattern.includes('*')) {
```

Comment:
On fresh install, this is throwing a background error `Cannot read property 'includes' of null`

Reply:
Looks to be coming from BrowserButton ln 154 `this._setIcon(!globals.SESSION.paused_blocking && !this.policy.checkSiteWhitelist(tab.url), tabId, trackerCount, alert);`

Reply:
Nice catch, I've added a check to ensure pattern is truthy